nixos/modules/services/networking/networkmanager.nix

```diff
@@ -227,13 +224,28 @@ in {
         '';
       };
 
-      packages = mkOption {
-        type = types.listOf types.package;
+      plugins = mkOption {
+        type =
+          let
+            networkManagerPluginPackage = types.package // {
+              description = "NetworkManager plug-in";
+              check =
+                p:
+                lib.assertMsg
+                  (types.package.check p
+                    && p ? networkManagerPlugin
+                    && lib.isString p.networkManagerPlugin)
+                  ''
+                    Package ‘${p.name}’, is not a NetworkManager plug-in.
+                    Those need to have a ‘networkManagerPlugin’ attribute.
+                  '';
+            };
+          in
+          types.listOf networkManagerPluginPackage;
         default = [ ];
```

Comment:
it might be helpful to add a defaultText listing the plugins added by default. if the docs say `Default: [ ]` we'd assume that no plugins would be loaded, which is not what's happening.

Reply:
I added a note to the description.

Reply:
This whole thing looks really off to me. Why have an option with no default if it effectively has one? If there is a default value, then shouldn't it be set at the option declaration location and not somewhere else?